src/android.ts

```diff
@@ -40,9 +41,24 @@ const BUTTON_MAP: Record<Button, string> = {
 const TIMEOUT = 30000;
 const MAX_BUFFER_SIZE = 1024 * 1024 * 4;
 
+type AndroidDeviceType = "tv" | "standard";
+
+type DpadButton = "DPAD_UP" | "DPAD_DOWN" | "DPAD_LEFT" | "DPAD_RIGHT" | "DPAD_CENTER";
+
 export class AndroidRobot implements Robot {
 
+	private deviceType: AndroidDeviceType = "standard"; // Default to standard
+
 	public constructor(private deviceId: string) {
+		// --- Device Type Detection ---
+		try {
+			const features = this.adb("shell", "pm", "list", "features").toString();
+			if (features.includes("android.software.leanback") || features.includes("android.hardware.type.television")) {
+				this.deviceType = "tv";
+			}
```

Comment:
`memo`
```sh
adb shell pm list features
```
If `android.software.leanback` is present then it is a sureshot way of identifying whether the device is Android TV or not.